legogroth16/src/generator.rs

```diff
@@ -165,15 +165,15 @@ where
 
     // Setup public params for the Subspace Snark
     let link_rows = 2; // we're comparing two commitments, proof.d and proof.link_d
-    let link_cols = commit_witness_count + 2; // we have `commit_witness_count` witnesses and 1 hiding factor per row
+    let link_cols = commit_witness_count as u64 + 2; // we have `commit_witness_count` witnesses and 1 hiding factor per row
```

Comment:
Accept `commit_witness_count` as `u32` arg in `generate_parameters_*` and `generate_random_parameters_*`